src/main/java/de/hpi/bpt/argos/api/eventQuery/EventQueryEndpointImpl.java

```diff
@@ -0,0 +1,198 @@
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String createEventQuery(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        try {
+            JsonObject jsonBody = jsonParser.parse(request.body()).getAsJsonObject();
+            JsonObject jsonEventQuery = jsonBody.get(JSON_EVENT_QUERY_ATTRIBUTE).getAsJsonObject();
+
+            if (jsonEventQuery == null) {
+                halt(HttpStatusCodes.BAD_REQUEST, "no event query given in body");
+            }
+
+            EventQuery eventQuery = null;
+            try {
+                eventQuery = new EventQueryImpl();
+                eventQuery.setTypeId(jsonEventQuery.get("EventTypeId").getAsLong());
+                eventQuery.setDescription(jsonEventQuery.get("Description").getAsString());
+                eventQuery.setQuery(jsonEventQuery.get("Query").getAsString());
+            } catch (Exception e) {
+                halt(HttpStatusCodes.BAD_REQUEST, "failed to parse event query");
+            }
+
+            EventPlatformFeedback feedback = EventProcessingPlatformUpdaterImpl.getInstance().registerEventQuery(eventQuery.getTypeId(), eventQuery);
+
+            if (!feedback.isSuccessful()) {
+                halt(HttpStatusCodes.ERROR, String.format("cannot register event type: %1$s", feedback.getResponseText()));
+            }
+
+            PersistenceAdapterImpl.getInstance().createArtifact(eventQuery, EventTypeEndpoint.getEventTypeQueriesUri(eventQuery.getId()));
+
+        } catch (HaltException halt) {
+            logger.info(String.format("cannot create event query: %1$s -> %2$s", halt.statusCode(), halt.body()));
+            throw halt;
+        } catch (Exception e) {
+            LoggerUtilImpl.getInstance().error(logger, "cannot parse request body to event query", e);
+            halt(HttpStatusCodes.ERROR, e.getMessage());
+        }
+        endpointUtil.logSendingResponse(logger, request, response.status(), response.body());
+        return "";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String deleteEventQuery(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        long eventQueryId = getEventQueryId(request);
+        EventQuery eventQuery = PersistenceAdapterImpl.getInstance().getEventQuery(eventQueryId);
+        if (eventQuery == null) {
+            halt(HttpStatusCodes.ERROR, "Given query was not found");
+        }
+
+        EventPlatformFeedback feedback = EventProcessingPlatformUpdaterImpl.getInstance().deleteEventQuery(eventQuery);
+        if (!feedback.isSuccessful()) {
+            halt(HttpStatusCodes.ERROR, String.format("cannot unregister event query: %1$s", feedback.getResponseText()));
+        }
+
+        if (!PersistenceAdapterImpl.getInstance().deleteArtifact(eventQuery,
+                EventTypeEndpoint.getEventTypeQueriesUri(eventQuery.getId()))) {

```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') NullPointerException might be thrown as 'eventQuery' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS2259)